javascript/code-challenges/challenges-11.test.js

```diff
@@ -0,0 +1,234 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named transformToLis that, given an object, returns an array of the key value pairs as html list items.
+
+For example:
+{
+  name: 'bob',
+  age: 32
+}
+
+Becomes:
+[
+<li>name: bob</li>,
+<li>age: 32</li>
+]
+------------------------------------------------------------------------------------------------ */
+
+function transformToLis(obj) {
+  return (obj.name && obj.age) ? ([`<li>name: ${obj.name}</li>`, `<li>age: ${obj.age}</li>`]) : [];
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named count that, given an integer and an array of arrays, uses either filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+
+Note: You might need to use the same method more than once.
+
+For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+------------------------------------------------------------------------------------------------ */
+
+const count = (target, input) => {
+  let numCount = 0;
+  input.map(item => item
+    .map(item => {
+      if (item === target) {
+        numCount++;
+      }
+    }));
+  return numCount;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that, given an array of integer arrays as input, calculates the total sum of all the elements in the array.
+
+You may want to use filter, map, or reduce for this problem, but are not required to. You may need to use the same method more than once.
+
+For example, [[1, 2, 3, 4, 5], [6, 7, 2, 4, 5, 7], [9, 2, 3, 6,]] returns 66.
+------------------------------------------------------------------------------------------------ */
+
+const totalSum = (input) => {
+  let sum = 0;
+  input.map(item => item
+    .map(item => sum += item));
+  return sum;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named divisibleByFiveTwoToThePower that accepts an array of arrays as input.
+
+This function should first remove any elements that are not numbers or are not divisible by five.
+
+This function should then raise 2 to the power of the resulting numbers, returning an array of arrays.
+
+For example, [ [0,2,5,4], [2,4,10], [] ] should return [ [1, 32], [1024], [] ].
+------------------------------------------------------------------------------------------------ */
+
+const divisibleByFiveTwoToThePower = (input) => {
+  let filteredInput = input.map(item => item
+    .filter(item => (typeof item === 'number' && item % 5 === 0)));
+  return filteredInput.map(item => item
+    .map(num => Math.pow(2, num)));
+};
```

Comment:
It really feels like you've begun to master your array methods. Really great work.